django/db/models/fields/files.py

```diff
@@ -272,7 +272,7 @@ def deconstruct(self):
         return name, path, args, kwargs
 
     def get_internal_type(self):
-        return "FileField"
+        return self.__class__.__name__
```

Comment:
This will unfortunately be backward compatible for any subclasses of `FileField` as the failures involving `ImageField` will likely demonstrate.

Django database backends use this return value to determine which data type they should be stored in

https://github.com/django/django/blob/f97bbad908df128189eff77d98af9a25ed1ecf23/django/db/backends/postgresql/base.py#L83

Since both `FileField` and `ImageField` are stored the same way, as `MEDIA_ROOT` relative paths, they both use the same `FileField` internal type.